### PR TITLE
feat: allow selecting alternate salt package repositories

### PR DESCRIPTION
**Pull request by major0**
Make the upstream salt package repository selectable, thus allowing the use of archived salt versions (hosted in http://archive.repo.saltproject.io), as well as custom salt versions hosted in alternate repositories.

### PR progress checklist (to be filled in by reviewers)

- [X] Changes to documentation are appropriate (or tick if not required)
- [X] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [X] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

**Review by n-rodriguez — APPROVED** [review]

File: pillar.example [hunk 1]
```diff
@@ -37,6 +37,10 @@ salt:
     salt_ssh: 'salt-ssh'
     pyinotify: 'python-pyinotify'  # the package to be installed for pyinotify
 
+  # Set which salt repository to use, default to https://repo.saltstack.com
+  # For older releases use http://archive.repo.saltproject.io
+  repo: 'https://archive.repo.saltproject.io'
```
Comment (n-rodriguez) [inline pillar.example:42]: It should be http to not break tests

**Review by n-rodriguez — CHANGES_REQUESTED** [review]

File: pillar.example (hunk 1 above)
Comment (major0) [reply pillar.example:42]: I suppose this is why the previous code had mixed http and https url's. :disappointed:

**Comment by myii** [comment]
@major0 Thanks for the PR.  You're not going to win either way, since all of the Debian/Ubuntu `pkgrepo` entries are using `http` and all of the `key_url` entries are using `https`.  Let's settle on one (probably `https`) and just fix the tests.  If you can make sure `Allow edits from maintainers` is allowed and I'll push the changes to complete this PR.

**Comment by major0** [comment]
> @major0 Thanks for the PR. You're not going to win either way, since all of the Debian/Ubuntu `pkgrepo` entries are using `http` and all of the `key_url` entries are using `https`. Let's settle on one (probably `https`) and just fix the tests. If you can make sure `Allow edits from maintainers` is allowed and I'll push the changes to complete this PR.

`Allow edits` is allowed.

**Review by myii — APPROVED** [review]

**Comment by myii** [comment]
Fixed and merged (to use `https`), @major0 -- thanks again for the PR.

Appreciate the review, @n-rodriguez.

**Pull request merged**